File: 92-reverse-linked-list-ii/reverse-linked-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseBetween(ListNode head, int left, int right) {
        ListNode dummy = new ListNode(0,head);
        ListNode prev = dummy;
        for(int i=0;i<left-1;i++){
            prev = prev.next;
        }
        ListNode p = prev;
        ListNode q = prev.next;
        ListNode curr = prev.next;
        for(int i=0;i<right-left+1;i++){
            ListNode n = curr.next;
            curr.next = prev;
            prev= curr;
            curr=n;
        }
        p.next= prev;
        q.next=curr;
        return dummy.next;
    }
    
}

File: 92-reverse-linked-list-ii/reverse-linked-list-ii.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ReverseBetween(ListNode head, int left, int right) {
        ListNode dummy = new ListNode(0,head);
        ListNode prev = dummy;
        for(int i=0;i<left-1;i++){
            prev = prev.next;
        }
        ListNode p = prev;
        ListNode q = prev.next;
        ListNode cur = q;
        for(int i=0;i<right-left+1;i++){
            ListNode n = cur.next;
            cur.next = prev;
            prev = cur;
            cur = n; 
        }
        p.next = prev;
        q.next = cur;
        return dummy.next;
    }
}